api/src/opentrons/protocol_engine/state/module_substates/heater_shaker_module_substate.py

```diff
@@ -64,3 +70,19 @@ def validate_target_speed(rpm: float) -> int:
                 f"Heater-Shaker got invalid speed of {rpm}RPM. Valid range is "
                 f"{HEATER_SHAKER_SPEED_RANGE}."
             )
+
+    def raise_if_labware_latch_not_closed(self) -> None:
+        """Raise an error if labware is not latched on the heater-shaker."""
+        if self.labware_latch_status != HeaterShakerLabwareLatchStatus.IDLE_CLOSED:
+            raise CannotPerformModuleAction(
+                f"Cannot perform the module action when labware latch open or in error."
+                f" Current latch status is: {self.labware_latch_status.value}"
+            )
+
+    def raise_if_shaking(self) -> None:
+        """Raise an error if the heater-shaker is currently shaking."""
+        if self.speed_status != SpeedStatus.IDLE:
+            raise CannotPerformModuleAction(
+                f"Cannot perform the module action when module is shaking or in error."
```

Comment:
Similar to above (this is the converse case)
```suggestion
                f"The Heater-Shaker can't open its labware latch while it is shaking."
```